--- common.old	2012-12-12 13:15:02.490430286 +0100
+++ common.h	2012-12-12 13:16:45.109869773 +0100
@@ -61,7 +61,7 @@
 #define NRPE_PACKET_VERSION_2   2               
 #define NRPE_PACKET_VERSION_1	1		/* older packet version identifiers (no longer supported) */
 
-#define MAX_PACKETBUFFER_LENGTH	1024		/* max amount of data we'll send in one query/response */
+#define MAX_PACKETBUFFER_LENGTH	8192		/* max amount of data we'll send in one query/response */
 											/* WARNING - do not change this as older clients/servers will not work */
 
 typedef struct packet_struct{
