--- heartbeat/conntrackd	2011-02-16 18:22:33.000000000 +0100
+++ heartbeat/conntrackd	2011-10-12 21:05:48.285711956 +0200
@@ -78,8 +78,8 @@
 <action name="demote"	timeout="30" />
 <action name="notify"	timeout="30" />
 <action name="stop"    timeout="30" />
-<action name="monitor" depth="0"  timeout="20" interval="20" role="Slave" />
-<action name="monitor" depth="0"  timeout="20" interval="10" role="Master" />
+<action name="monitor" timeout="20" interval="20" role="Slave" />
+<action name="monitor" timeout="20" interval="10" role="Master" />
 <action name="meta-data"  timeout="5" />
 <action name="validate-all"  timeout="30" />
 </actions>
@@ -111,8 +111,10 @@
 
 conntrackd_monitor() {
 	rc=$OCF_NOT_RUNNING
-	# It does not write a PID file, so check with pgrep
-	pgrep -f $OCF_RESKEY_binary && rc=$OCF_SUCCESS
+	# It does not write a PID file, so check the socket exists after
+	# extracting its path from the configuration file
+	local conntrack_socket=$(awk '/^[ \t]*UNIX[ \t]*{/,/^[ \t]*}/ { if ($1 == "Path") { print $2 } }' $OCF_RESKEY_config)
+	[ -S "$conntrack_socket" ] && rc=$OCF_SUCCESS
 	if [ "$rc" -eq "$OCF_SUCCESS" ]; then
 		# conntrackd is running 
 		# now see if it acceppts queries
