diff -urN stunnel-4.41/src/options.c stunnel-4.41-new//src/options.c
--- stunnel-4.41/src/options.c	2011-08-01 22:20:02.000000000 +0200
+++ stunnel-4.41-new//src/options.c	2011-08-01 22:26:16.321610738 +0200
@@ -1533,6 +1533,24 @@
         break;
     }
 
+    /* listenqueue */
+    switch(cmd) {
+    case CMD_INIT:
+        section->listenqueue=SOMAXCONN;
+        break;
+    case CMD_EXEC:
+        if(strcasecmp(opt, "listenqueue"))
+            break;
+        section->listenqueue=atoi(arg);
+        return (section->listenqueue?NULL:"Bad verify level");
+    case CMD_DEFAULT:
+        s_log(LOG_NOTICE, "%-15s = %d", "listenqueue", SOMAXCONN);
+        break;
+    case CMD_HELP:
+        s_log(LOG_NOTICE, "%-15s = defines the maximum length the queue of pending connections may grow to", "listenqueue");
+        break;
+    }
+
     if(cmd==CMD_EXEC)
         return option_not_found;
     return NULL; /* OK */
diff -urN stunnel-4.41/src/prototypes.h stunnel-4.41-new//src/prototypes.h
--- stunnel-4.41/src/prototypes.h	2011-08-01 22:20:02.000000000 +0200
+++ stunnel-4.41-new//src/prototypes.h	2011-08-01 22:26:54.341285560 +0200
@@ -165,6 +165,7 @@
     int timeout_close;                          /* maximum close_notify time */
     int timeout_connect;                           /* maximum connect() time */
     int timeout_idle;                        /* maximum idle connection time */
+    int listenqueue;                                        /* Listen baklog */
     enum {FAILOVER_RR, FAILOVER_PRIO} failover;         /* failover strategy */
 
         /* service-specific data for protocol.c */
diff -urN stunnel-4.41/src/stunnel.c stunnel-4.41-new//src/stunnel.c
--- stunnel-4.41/src/stunnel.c	2011-08-01 22:20:02.000000000 +0200
+++ stunnel-4.41-new//src/stunnel.c	2011-08-01 22:27:47.030780042 +0200
@@ -248,7 +248,7 @@
             }
             s_log(LOG_DEBUG, "Service %s bound to %s",
                 opt->servname, local_address);
-            if(listen(opt->fd, SOMAXCONN)) {
+            if(listen(opt->fd, opt->listenqueue)) {
                 sockerror("listen");
                 closesocket(opt->fd);
                 return 1;
