--- sample_config/feed_passive/Makefile.in~
+++ sample_config/feed_passive/Makefile.in
@@ -25,9 +25,9 @@
 
 install-config: 
 	$(PERL) gencfg 10
-	$(INSTALL) -m 755 $(INSTALL_OPTS) -d ${CFGDIR}/feed_passive
-	$(INSTALL) -m 755 $(INSTALL_OPTS) addhosts.cfg ${CFGDIR}/feed_passive
-	$(INSTALL) -m 755 $(INSTALL_OPTS) gencfg ${CFGDIR}/feed_passive
-	$(INSTALL) -m 755 $(INSTALL_OPTS) feed_passive.cmd ${CFGDIR}/feed_passive
-	$(INSTALL) -m 755 $(INSTALL_OPTS) feed_passive.cfg ${CFGDIR}/feed_passive
-	$(INSTALL) -m 755 $(INSTALL_OPTS) service_definition.tpl ${CFGDIR}/feed_passive
+	$(INSTALL) -m 755 $(INSTALL_OPTS) -d $(DESTDIR)${CFGDIR}/feed_passive
+	$(INSTALL) -m 755 $(INSTALL_OPTS) addhosts.cfg $(DESTDIR)${CFGDIR}/feed_passive
+	$(INSTALL) -m 755 $(INSTALL_OPTS) gencfg $(DESTDIR)${CFGDIR}/feed_passive
+	$(INSTALL) -m 755 $(INSTALL_OPTS) feed_passive.cmd $(DESTDIR)${CFGDIR}/feed_passive
+	$(INSTALL) -m 755 $(INSTALL_OPTS) feed_passive.cfg $(DESTDIR)${CFGDIR}/feed_passive
+	$(INSTALL) -m 755 $(INSTALL_OPTS) service_definition.tpl $(DESTDIR)${CFGDIR}/feed_passive
