--- a/Makefile
+++ b/Makefile
@@ -44,8 +44,10 @@
 
 config:
 	sh -c 'type xmkmf'
+ifneq ($(JSRC),)
 	sh -c 'type javac'
 	sh -c 'type jar'
+endif
 	@echo
 	cd $(VSRC)/libvncauth; pwd; xmkmf
 	cd $(VSRC)/vncviewer;  pwd; xmkmf
@@ -55,13 +55,17 @@ config:
 all:
 	cd $(VSRC)/libvncauth; $(MAKE)
 	cd $(VSRC)/vncviewer;  $(MAKE)
+ifneq ($(JSRC),)
 	cd $(JSRC); $(MAKE)
+endif
 	cd $(PSRC); $(MAKE)
 
 clean:
 	cd $(VSRC)/libvncauth; $(MAKE) clean
 	cd $(VSRC)/vncviewer;  $(MAKE) clean
+ifneq ($(JSRC),)
 	cd $(JSRC); $(MAKE) clean
+endif
 	cd $(PSRC); $(MAKE) clean
 
 install: all
@@ -72,7 +76,9 @@ install: all
 	cp -p $(LIMACC) $(ROOT)$(PREFIX)/$(LIB)
 	cp -p $(ULTDSM) $(ROOT)$(PREFIX)/$(LIB)
 	cp -pR scripts/* $(ROOT)$(PREFIX)/$(LIB)
+ifneq ($(JSRC),)
 	cp -p $(ARCHIVE) $(ROOT)$(PREFIX)/$(LIB)/util
+endif
 	cp -p $(MAN)/man1/ssvnc.1 $(ROOT)$(PREFIX)/$(MANDIR)/man1
 	./wr_tool $(ROOT)$(PREFIX)/$(BIN)/ssvnc  $(PREFIX)/$(LIB)/ssvnc
 	./wr_tool $(ROOT)$(PREFIX)/$(BIN)/tsvnc  $(PREFIX)/$(LIB)/tsvnc
