--- plugins/mk_inventory.linux
+++ plugins/mk_inventory.linux
@@ -44,6 +44,12 @@
         dpkg-query --show --showformat='${Package}|${Version}|${Architecture}|deb|${Summary}|${Status}\n'
     fi
 
+    # List Gentoo packages
+    if type equery >/dev/null; then
+        echo "<<<lnx_packages:sep(124):persist($UNTIL)>>>"
+        equery -C list --format '$category/$name|$fullversion|$mask2|ebuild|Repository $repo|installed' \* | head -n -1
+    fi
+
     # List of RPM packages in same format
     if type rpm >/dev/null; then
         echo "<<<lnx_packages:sep(9):persist($UNTIL)>>>"
@@ -52,7 +58,7 @@
 
     # Information about distribution
     echo "<<<lnx_distro:sep(124):persist($UNTIL)>>>"
-    for f in {/etc/{debian_version,lsb-release,redhat-release,SuSE-release},/usr/share/cma/version} ; do
+    for f in {/etc/{debian_version,gentoo-release,lsb-release,redhat-release,SuSE-release},/usr/share/cma/version} ; do
         if [ -e $f ] ; then
             echo -n "$f|" ; tr \\n \| < $f | sed 's/|$//' ; echo
         fi
