diff --git a/src/obu.c b/src/obu.c
index 89fa593..ebce928 100644
--- a/src/obu.c
+++ b/src/obu.c
@@ -1291,7 +1291,7 @@ int dav1d_parse_obus(Dav1dContext *const c, Dav1dData *const in, const int globa
     case DAV1D_OBU_FRAME:
     case DAV1D_OBU_FRAME_HDR:
         if (global) break;
-        if (!c->seq_hdr) goto error;
+        if (!c->seq_hdr) return len + init_byte_pos;
         if (!c->frame_hdr_ref) {
             c->frame_hdr_ref = dav1d_ref_create_using_pool(c->frame_hdr_pool,
                                                            sizeof(Dav1dFrameHeader));
