--- check_mk-1.2.2p1.orig/setup.sh	2013-04-30 16:51:30.000000000 +0200
+++ check_mk-1.2.2p1/setup.sh	2013-07-02 15:41:12.324034278 +0200
@@ -550,6 +550,8 @@
    mkdir -p $D
    tar xvzf $SRCDIR/livestatus.tar.gz -C $D
    pushd $D
+   # remove strip from Makefile.in
+   sed -i -e 's/^STRIP = @STRIP@$/STRIP = \/bin\/true/' Makefile.in
    ./configure --libdir=$libdir --bindir=$bindir &&
    make clean &&
    cat <<EOF > src/livestatus.h &&
@@ -559,7 +561,8 @@
 #endif // livestatus_h
 EOF
    make -j 8  2>&1 &&
-   strip src/livestatus.o &&
+   # remove strip to get meaningful backtraces
+   #strip src/livestatus.o &&
    mkdir -p $DESTDIR$libdir &&
    install -m 755 src/livecheck src/livestatus.o $DESTDIR$libdir &&
    mkdir -p $DESTDIR$bindir &&
@@ -993,11 +996,12 @@
 	   done &&
            # make htpasswd writable by apache, since we need this for
            # WATO. Also create an empty and Apache-writable auth.serials
-           serials_file=${htpasswd_file%/*}/auth.serials &&
+           serials_file=$DESTDIR${htpasswd_file%/*}/auth.serials &&
            touch "$serials_file" &&
            chown $wwwuser "$serials_file" &&
            chown $wwwuser "$htpasswd_file" &&
-	   create_sudo_configuration &&
+	   # disabled modification of /etc/sudoers because of ebuild access violation
+	   #create_sudo_configuration &&
            if [ "$enable_mkeventd" = yes ]
            then
 	       if [ -z "$YES" ] ; then echo -n "(Compiling Event Console binaries..." ; fi
