Index: pdns/mastercommunicator.cc
===================================================================
--- pdns/mastercommunicator.cc	(revision 2343)
+++ pdns/mastercommunicator.cc	(working copy)
@@ -151,7 +151,7 @@
     if(p.d.rcode)
       L<<Logger::Warning<<"Received unsuccessful notification report for '"<<p.qdomain<<"' from "<<from.toStringWithPort()<<", rcode: "<<p.d.rcode<<endl;      
     
-    if(d_nq.removeIf(from.toStringWithPort(), p.d.id, p.qdomain))
+    if(d_nq.removeIf(from.toString(), p.d.id, p.qdomain))
       L<<Logger::Warning<<"Removed from notification list: '"<<p.qdomain<<"' to "<<from.toStringWithPort()<< (p.d.rcode ? "" : " (was acknowledged)")<<endl;      
     else {
       L<<Logger::Warning<<"Received spurious notify answer for '"<<p.qdomain<<"' from "<< from.toStringWithPort()<<endl;
Index: pdns/communicator.hh
===================================================================
--- pdns/communicator.hh	(revision 2343)
+++ pdns/communicator.hh	(working copy)
@@ -82,10 +82,7 @@
       //      cout<<i->id<<" "<<id<<endl;
       //cout<<i->ip<<" "<<remote<<endl;
       //cout<<i->domain<<" "<<domain<<endl;
-      string remoteIP, ourIP, port;
-      tie(remoteIP, port)=splitField(remote,':');
-      tie(ourIP, port)=splitField(i->ip,':');
-      if(i->id==id && ourIP == remoteIP && i->domain==domain) {
+      if(i->id==id && i->ip == remote && i->domain==domain) {
         d_nqueue.erase(i);
         return true;
       }
