commit abc5986b89cc76c23d52524686d1e13e2360926b
Author: frainz-de <paul.michel@mailbox.org>
Date:   Wed Nov 22 21:58:29 2017 +0100

    Fix compilation with gcc >= 6

diff --git a/SmartCardServices/src/PKCS11dotNetV2/CardModuleService.hpp b/SmartCardServices/src/PKCS11dotNetV2/CardModuleService.hpp
index 2c73710..bc297f0 100644
--- a/SmartCardServices/src/PKCS11dotNetV2/CardModuleService.hpp
+++ b/SmartCardServices/src/PKCS11dotNetV2/CardModuleService.hpp
@@ -25,12 +25,12 @@
 
 #include <string>
 #include <boost/ptr_container/ptr_map.hpp>
+#include "Log.hpp"
 #include "MarshallerCfg.h"
 #include "Array.hpp"
 #include "Marshaller.h"
 #include "Except.h"
 #include "Timer.hpp"
-#include "Log.hpp"
 
 
 const unsigned char CARD_FREE_SPACE  = 0x00; //Returns a byte array blob of 12 bytes
diff --git a/SmartCardServices/src/PKCS11dotNetV2/MiniDriverContainer.hpp b/SmartCardServices/src/PKCS11dotNetV2/MiniDriverContainer.hpp
index 7ca1ee2..a4c8f40 100644
--- a/SmartCardServices/src/PKCS11dotNetV2/MiniDriverContainer.hpp
+++ b/SmartCardServices/src/PKCS11dotNetV2/MiniDriverContainer.hpp
@@ -29,8 +29,8 @@
 #include <boost/serialization/version.hpp>
 #include <boost/shared_ptr.hpp>
 #include "Array.hpp"
-#include "cardmod.h"
 #include "Log.hpp"
+#include "cardmod.h"
 #include "MiniDriverAuthentication.hpp"
 
 
diff --git a/SmartCardServices/src/PKCS11dotNetV2/MiniDriverContainerMapFile.cpp b/SmartCardServices/src/PKCS11dotNetV2/MiniDriverContainerMapFile.cpp
index 66b5f5a..a233fe7 100644
--- a/SmartCardServices/src/PKCS11dotNetV2/MiniDriverContainerMapFile.cpp
+++ b/SmartCardServices/src/PKCS11dotNetV2/MiniDriverContainerMapFile.cpp
@@ -22,10 +22,8 @@
 #include <cstdio>
 #include <boost/foreach.hpp>
 #include <memory>
-#include "cardmod.h"
 #include "MiniDriverContainerMapFile.hpp"
 #include "MiniDriverFiles.hpp"
-#include "Log.hpp"
 #include "Timer.hpp"
 #include "MiniDriverException.hpp"
 #include "sha1.h"
diff --git a/SmartCardServices/src/PKCS11dotNetV2/MiniDriverContainerMapFile.hpp b/SmartCardServices/src/PKCS11dotNetV2/MiniDriverContainerMapFile.hpp
index c43de24..8f6aee6 100644
--- a/SmartCardServices/src/PKCS11dotNetV2/MiniDriverContainerMapFile.hpp
+++ b/SmartCardServices/src/PKCS11dotNetV2/MiniDriverContainerMapFile.hpp
@@ -29,9 +29,10 @@
 #include <boost/array.hpp>
 #include <boost/shared_ptr.hpp>
 #include <string>
-#include "MiniDriverContainer.hpp"
+#include "Log.hpp"
 #include "Array.hpp"
 #include "CardModuleService.hpp"
+#include "MiniDriverContainer.hpp"
 
 
 const int g_MaxContainer = 15;
