Index: MiniDriverAuthentication.hpp
===================================================================
--- MiniDriverAuthentication.hpp	(revision 160)
+++ MiniDriverAuthentication.hpp	(working copy)
@@ -23,6 +23,8 @@
 #define __GEMALTO_MINIDRIVER_AUTHENTICATION__
 
 
+#include <boost/serialization/extended_type_info.hpp>
+#include <boost/serialization/singleton.hpp>
 #include <boost/serialization/serialization.hpp>
 #include <boost/serialization/shared_ptr.hpp>
 #include <boost/shared_ptr.hpp>
Index: MiniDriverContainerMapFile.hpp
===================================================================
--- MiniDriverContainerMapFile.hpp	(revision 160)
+++ MiniDriverContainerMapFile.hpp	(working copy)
@@ -23,6 +23,8 @@
 #define __GEMALTO_MINIDRIVER_CONTAINER_MAP_FILE__
 
 
+#include <boost/serialization/singleton.hpp>
+#include <boost/serialization/extended_type_info.hpp>
 #include <boost/serialization/serialization.hpp>
 #include <boost/serialization/shared_ptr.hpp>
 #include <boost/serialization/array.hpp>
Index: MiniDriverFiles.hpp
===================================================================
--- MiniDriverFiles.hpp	(revision 160)
+++ MiniDriverFiles.hpp	(working copy)
@@ -26,6 +26,8 @@
 #include <boost/ptr_container/serialize_ptr_map.hpp>
 #include <boost/ptr_container/ptr_map.hpp>
 #include <boost/serialization/string.hpp>
+#include <boost/serialization/singleton.hpp>
+#include <boost/serialization/extended_type_info.hpp>
 #include <boost/serialization/vector.hpp>
 #include <boost/serialization/set.hpp>
 #include <boost/serialization/shared_ptr.hpp>
