--- php-src/php.ini-dist.orig  2010-03-19 11:38:24.000000000 +0100
+++ php-src/php.ini-dist       2010-03-19 11:38:50.000000000 +0100
@@ -714,6 +714,12 @@
 ; the 5th parameter to mail(), even in safe mode.
 ;mail.force_extra_parameters =
 
+; Add X-PHP-Originaiting-Script: that will include uid of the script followed by the filename
+mail.add_x_header = Off
+
+; Log all mail() calls including the full path of the script, line #, to address and headers
+mail.log =
+
 [SQL]
 sql.safe_mode = Off

--- php-src/php.ini-recommended.orig  2010-03-19 11:38:24.000000000 +0100
+++ php-src/php.ini-recommended       2010-03-19 11:38:50.000000000 +0100
@@ -765,6 +765,12 @@
 ; the 5th parameter to mail(), even in safe mode.
 ;mail.force_extra_parameters =
 
+; Add X-PHP-Originaiting-Script: that will include uid of the script followed by the filename
+mail.add_x_header = On
+
+; Log all mail() calls including the full path of the script, line #, to address and headers
+mail.log =
+
 [SQL]
 sql.safe_mode = Off
 
--- php-src/main/main.c.orig  2010-03-19 11:38:24.000000000 +0100
+++ php-src/main/main.c       2010-03-19 11:38:50.000000000 +0100
@@ -445,6 +445,8 @@
 
 	PHP_INI_ENTRY("SMTP",						"localhost",PHP_INI_ALL,		NULL)
 	PHP_INI_ENTRY("smtp_port",					"25",		PHP_INI_ALL,		NULL)
+	STD_PHP_INI_BOOLEAN("mail.add_x_header",			"0",		PHP_INI_SYSTEM|PHP_INI_PERDIR,		OnUpdateBool,			mail_x_header,			php_core_globals,	core_globals)
+	STD_PHP_INI_ENTRY("mail.log",					NULL,		PHP_INI_SYSTEM|PHP_INI_PERDIR,		OnUpdateString,			mail_log,			php_core_globals,	core_globals)
 	PHP_INI_ENTRY("browscap",					NULL,		PHP_INI_SYSTEM,		NULL)
 	PHP_INI_ENTRY("memory_limit",				"128M",		PHP_INI_ALL,		OnChangeMemoryLimit)
 	PHP_INI_ENTRY("precision",					"14",		PHP_INI_ALL,		OnSetPrecision)


--- php-src/main/php_globals.h.orig  2010-03-19 11:38:24.000000000 +0100
+++ php-src/main/php_globals.h       2010-03-19 11:38:50.000000000 +0100
@@ -162,6 +162,9 @@
 	long max_input_nesting_level;
 	zend_bool in_user_include;
 	zend_bool in_error_log;
+
+	zend_bool mail_x_header;
+	char *mail_log;
 };
 

--- php-src/ext/standard/mail.c.orig	2010-03-19 12:08:53.000000000 +0100
+++ php-src/ext/standard/mail.c	2010-03-19 12:18:47.000000000 +0100
@@ -23,6 +23,8 @@
 #include <stdio.h>
 #include "php.h"
 #include "ext/standard/info.h"
+#include "ext/standard/php_string.h"
+#include "ext/standard/basic_functions.h"
 
 #if HAVE_SYSEXITS_H
 #include <sysexits.h>
@@ -67,6 +69,7 @@
 		*p = ' ';				\
 	}						\
 
+extern long php_getuid(void);
 
 /* {{{ proto int ezmlm_hash(string addr)
    Calculate EZMLM list hash value. */
@@ -206,25 +209,62 @@
 	int ret;
 	char *sendmail_path = INI_STR("sendmail_path");
 	char *sendmail_cmd = NULL;
+	char *mail_log = INI_STR("mail.log");
+	char *hdr = headers;
+
 #if PHP_SIGCHILD
 	void (*sig_handler)() = NULL;
 #endif
 
+#define MAIL_RET(val)		\
+	if (hdr != headers) {	\
+		efree(hdr);	\
+	}			\
+	return val;		\
+
+	if (mail_log) {
+		char *tmp;
+		int l = spprintf(&tmp, 0, "mail() on [%s:%d]: To: %s -- Headers: %s\n", zend_get_executed_filename(TSRMLS_C), zend_get_executed_lineno(TSRMLS_C), to, hdr ? hdr : "");
+		if (hdr) { /* find all \r\n instances and replace them with spaces, so a log line is always one line long */
+			char *p = tmp;
+			while ((p = strpbrk(p, "\r\n"))) {
+				*p = ' ';
+			}
+			tmp[l - 1] = '\n';
+		}
+		_php_error_log(3, tmp, mail_log, NULL TSRMLS_CC);
+		efree(tmp);
+	}
+	if (PG(mail_x_header)) {
+		char *tmp = zend_get_executed_filename(TSRMLS_C);
+		char *f;
+		size_t f_len;
+
+		php_basename(tmp, strlen(tmp), NULL, 0,&f, &f_len);
+
+		if (headers != NULL) {
+			spprintf(&hdr, 0, "%s\r\nX-PHP-Originating-Script: %ld:%s\n", headers, php_getuid(), f);
+		} else {
+			spprintf(&hdr, 0, "X-PHP-Originating-Script: %ld:%s\n", php_getuid(), f);
+		}
+		efree(f);
+	}
+
 	if (!sendmail_path) {
 #if (defined PHP_WIN32 || defined NETWARE)
 		/* handle old style win smtp sending */
-		if (TSendMail(INI_STR("SMTP"), &tsm_err, &tsm_errmsg, headers, subject, to, message, NULL, NULL, NULL TSRMLS_CC) == FAILURE) {
+		if (TSendMail(INI_STR("SMTP"), &tsm_err, &tsm_errmsg, hdr, subject, to, message, NULL, NULL, NULL TSRMLS_CC) == FAILURE) {
 			if (tsm_errmsg) {
 				php_error_docref(NULL TSRMLS_CC, E_WARNING, "%s", tsm_errmsg);
 				efree(tsm_errmsg);
 			} else {
 				php_error_docref(NULL TSRMLS_CC, E_WARNING, "%s", GetSMErrorText(tsm_err));
 			}
-			return 0;
+			MAIL_RET(0);
 		}
-		return 1;
+		MAIL_RET(1);
 #else
-		return 0;
+		MAIL_RET(0);
 #endif
 	}
 	if (extra_cmd != NULL) {
@@ -267,13 +307,13 @@
 				signal(SIGCHLD, sig_handler);
 			}
 #endif	
-			return 0;
+			MAIL_RET(0);
 		}
 #endif
 		fprintf(sendmail, "To: %s\n", to);
 		fprintf(sendmail, "Subject: %s\n", subject);
-		if (headers != NULL) {
-			fprintf(sendmail, "%s\n", headers);
+		if (hdr != NULL) {
+			fprintf(sendmail, "%s\n", hdr);
 		}
 		fprintf(sendmail, "\n%s\n", message);
 		ret = pclose(sendmail);
@@ -295,9 +335,9 @@
 #endif
 #endif
 		{
-			return 0;
+			MAIL_RET(0);
 		} else {
-			return 1;
+			MAIL_RET(1);
 		}
 	} else {
 		php_error_docref(NULL TSRMLS_CC, E_WARNING, "Could not execute mail delivery program '%s'", sendmail_path);
@@ -306,10 +346,10 @@
 			signal(SIGCHLD, sig_handler);						
 		}
 #endif
-		return 0;
+		MAIL_RET(0);
 	}
 
-	return 1; /* never reached */
+	MAIL_RET(1); /* never reached */
 }
 /* }}} */
 
