--- ext/odbc/config.m4.orig	2010-09-16 19:13:51.890526877 +0200
+++ ext/odbc/config.m4	2010-09-16 19:17:49.295529964 +0200
@@ -401,9 +401,9 @@
   fi
   if test "$withval" != "no"; then
     PHP_ADD_LIBRARY_WITH_PATH(iodbc, $withval/lib)
-    PHP_ADD_INCLUDE($withval/include, 1)
+    PHP_ADD_INCLUDE($withval/include/iodbc, 1)
     ODBC_TYPE=iodbc
-    ODBC_INCLUDE=-I$withval/include
+    ODBC_INCLUDE=-I$withval/include/iodbc
     ODBC_LFLAGS=-L$withval/lib
     ODBC_LIBS=-liodbc
     AC_DEFINE(HAVE_IODBC,1,[ ])
