--- src/spectrum_util.cpp
+++ src/spectrum_util.cpp
@@ -28,6 +28,7 @@
 #include "protocols/abstractprotocol.h"
 #include "transport.h"
 #include <sys/param.h>
+#include "unistd.h"
 #ifdef BSD
 #include <sys/types.h>
 #include <sys/sysctl.h>
--- src/configinterface.cpp
+++ src/configinterface.cpp
@@ -24,6 +24,7 @@
 #include "log.h"
 #include "errno.h"
 #include "transport.h"
+#include "unistd.h"
 #include "abstractconfiginterfacehandler.h"
 #include "adhoc/adhocadmin.h"
 
