--- bontmia	2012-03-28 00:24:48.846074606 +0200
+++ bontmia	2012-03-28 00:29:53.408377248 +0200
@@ -1,4 +1,4 @@
-#!/bin/bash
+#!/usr/bin/env bash
 #
 # bontmia (Backup Over Network To Multiple Incremental Archives)
 #
@@ -8,6 +8,11 @@
 #
 # Written by John Enok Vollestad in April 2003 and have later
 # undergone some bugfixes and enhancements.
+#
+# Copyright (c) 2012 Marcel Pennewiss (opensource@pennewiss.de)
+#
+# Changelog:
+#   0.14-r1 - adding identity (ssh key) and exclude-from option
 
 print_usage()
 {
@@ -15,7 +20,8 @@
 
 NAME
         Bontmia - Backup Over Network To Multiple Incremental Archives
-        Version 0.14
+        Version 0.14 - Release 1
+	
 
 SYNOPSIS
         bontmia --dest <dest. dir> [options] [source dir [source dir ...]]
@@ -120,7 +126,13 @@
                 Specifies the port number to connect to on the remote
                 host when using ssh, which is the only option.
 
-
+	--identity <file>
+		Specifies the private key file for ssh
+	
+	--exclude-from <file>
+		Specifies the file for rsync containing the folders
+		which don't want to be synchronized
+		
 EXAMPLES
         bontmia --dest ./backup  --rotation \\
                 5minutes0hours0days0weeks0month0years \\
@@ -180,6 +192,8 @@
 unlock_destination()
 {
     rm -f "${tmpdir}/is_running.lock"
+    rm -f /tmp/bontmia_rsync_output.$$ > /dev/null 2>&1
+    rm -f /tmp/bontmia_rsync_output.$$.2 > /dev/null 2>&1
 }
 
 
@@ -349,15 +363,14 @@
 
 	if test "x$dryrun" == "xno"; then
 	    mkdir -p "$tmpdir/unfinished_backup/$this_backup/$hostname"
-
-	    rsync ${rsync_options} -e "ssh -p $port" "${dir}" "${tmpdir}/unfinished_backup/$this_backup/${hostname}" 2>&1 >/tmp/bontmia_rsync_output.$$ || {
+	    rsync ${rsync_options} --rsync-path='sudo /usr/bin/rsync' -e "ssh -p $port $identity" "${dir}/" "${tmpdir}/unfinished_backup/$this_backup/${hostname}" 2>&1 >/tmp/bontmia_rsync_output.$$ || {
 		echo
 		echo "  Caught an error doing rsync (return code $?)"
 		echo "  The last 10 lines of output from rsync:"
 		tail -10 /tmp/bontmia_rsync_output.$$
 		echo
 		echo "  Retrying rsync..."
-		rsync ${rsync_options} -e "ssh -p $port" "${dir}" "${tmpdir}/unfinished_backup/$this_backup/${hostname}" 2>&1 >/tmp/bontmia_rsync_output.$$ || {
+		rsync ${rsync_options} --rsync-path='sudo /usr/bin/rsync' -e "ssh -p $port $identity" "${dir}/" "${tmpdir}/unfinished_backup/$this_backup/${hostname}" 2>&1 >/tmp/bontmia_rsync_output.$$.2 || {
 		    echo
 		    echo "  Still no luck.  Rsync failed with returncode $?"
 		    echo "  $dir"
@@ -600,6 +613,8 @@
 exit_status="0"
 do_del_old="no"
 port="22"
+identity=""
+excludefrom=""
 compression=""
 rotation=""
 dryrun="no"
@@ -634,6 +649,14 @@
 	    shift
 	    port="$1"
 	    shift;;
+	( "--identity" )
+	    shift
+	    identity="-i $1"
+	    shift;;
+	( "--exclude-from" )
+	    shift
+	    excludefrom="--exclude-from=$1"
+	    shift;;
 	( "--bwlimit" )
 	    shift
 	    bwlimit="--bwlimit=$1"
@@ -696,7 +719,8 @@
 check_for_programs
 
 tmpdir=${tmpdir:-"$backup_destination"}
-rsync_options="-azv -T $tmpdir --force --relative --hard-links --delete $bwlimit" # --stats"
+
+rsync_options="-azv -T $tmpdir --force --relative --hard-links --delete $bwlimit $excludefrom" # --stats"
 
 # to speed up checking for files outside the backup areas
 bdirmatch=$(
