--- a/Makefile	2008-11-10 14:56:55.000000000 +0100
+++ b/Makefile	2009-04-24 18:02:18.000000000 +0200
@@ -1,20 +1,38 @@
 CC = gcc
-CFLAGS = -O2 -Wall -I /usr/include/PCSC/
+DESTDIR?=
+CFLAGS += -Wall -I /usr/include/PCSC/ -fPIC -D_REENTRANT
+
+# chose from
+#   USE_PORT_BASE1 (Matrica: Moneyplex)
+#     Use Portnumber to select smartcard reader, lowest input is 1
+#     Moneyplex: select COMX to select the Xth reader
+#   USE_PORT_BASE0 (any)
+#     Use Portnumber to select smartcard reader, lowest input is 0
+#   USE_CTN_BASE1 (any)
+#     Use CTN to select smartcard reader, lowest input is 1
+#   USE_CTN_BASE0 (jameica / libjavacard)
+#     Use CTN to select smartcard reader, lowest input is 0
+#     jameica: enter X into "Index des Lesers"/"Index of Reader"
+#              to select Xth smartcard reader
+# else:
+#   use the last reader returned by pcsc
 
 pcsc-ctapi-wrapper: pcsc-ctapi-wrapper.o
-	$(CC) $(CFLAGS) -shared -o libpcsc-ctapi-wrapper.so.0.3 -Wl,-soname="libpcsc-ctapi-wrapper.so" pcsc-ctapi-wrapper.o -lpcsclite
+	$(CC) $(CFLAGS) -shared -o libpcsc-ctapi-wrapper.so.0.4 -Wl,-soname="libpcsc-ctapi-wrapper.so",-z,defs pcsc-ctapi-wrapper.o -lpcsclite
+	strip --strip-unneeded pcsc-ctapi-wrapper.o
 
 pcsc-ctapi-wrapper.o: pcsc-ctapi-wrapper.c
 	$(CC) $(CFLAGS) -c -fPIC pcsc-ctapi-wrapper.c
 
 clean:
-	rm libpcsc-ctapi-wrapper.so.0.3 pcsc-ctapi-wrapper.o
+	rm -f libpcsc-ctapi-wrapper.so.0.4 pcsc-ctapi-wrapper.o
 
 install: pcsc-ctapi-wrapper
-	cp libpcsc-ctapi-wrapper.so.0.3 /usr/local/lib
-	ldconfig
+	install -d $(DESTDIR)/usr/lib
+	install -m644 libpcsc-ctapi-wrapper.so.0.4 $(DESTDIR)/usr/lib
+	ldconfig -l $(DESTDIR)/usr/lib/libpcsc-ctapi-wrapper.so.0.4
 
 uninstall: pcsc-ctapi-wrapper
-	rm /usr/local/lib/libpcsc-ctapi-wrapper.so
-	rm /usr/local/lib/libpcsc-ctapi-wrapper.so.0.3
+	rm $(DESTDIR)/usr/lib/libpcsc-ctapi-wrapper.so
+	rm $(DESTDIR)/usr/lib/libpcsc-ctapi-wrapper.so.0.4
 	ldconfig
