--- a/pcsc-ctapi-wrapper.c	2008-11-10 14:56:21.000000000 +0100
+++ b/pcsc-ctapi-wrapper.c	2009-04-24 18:02:18.000000000 +0200
@@ -176,7 +176,18 @@
   	    {
 		DEBUG_MESSAGE("CT_init: processing \"%s\", counter=%d \n",ptr,nbReaders);
 		nbReaders++;
+#ifdef USE_PORT_BASE1
 		if (nbReaders == pn) break;
+#endif
+#ifdef USE_PORT_BASE0
+		if (nbReaders-1 == pn) break;
+#endif
+#ifdef USE_CTN_BASE1
+		if (nbReaders == ctn) break;
+#endif
+#ifdef USE_CTN_BASE0
+		if (nbReaders-1 == ctn) break;
+#endif
 		ptr += strlen(ptr)+1;
             }
 	}   
