--- ffmpeg-1.0/libavformat/utils.c	2012-09-28 03:37:35.000000000 +0200
@@ -43,6 +43,8 @@
 #include "audiointerleave.h"
 #include "url.h"
 #include <stdarg.h>
+#undef time
+#include <time.h>
 #if CONFIG_NETWORK
 #include "network.h"
 #endif
@@ -4093,12 +4095,15 @@
                           const char *path, int number)
 {
     const char *p;
-    char *q, buf1[20], c;
-    int nd, len, percentd_found;
+    char *q, buf1[64], c;
+    int nd, len, percentd_found, percentt_found;
+    struct tm* ptm;
+    time_t epoch;
 
     q = buf;
     p = path;
     percentd_found = 0;
+    percentt_found = 0;
     for(;;) {
         c = *p++;
         if (c == '\0')
@@ -4126,6 +4131,21 @@
                 memcpy(q, buf1, len);
                 q += len;
                 break;
+            case 't':
+                if (percentt_found)
+                    goto fail;
+                percentt_found = 1;
+                // get current time as string and put it into the buffer
+                time(&epoch);
+                ptm = localtime(&epoch);
+                buf1[63] = (char) 0;
+                strftime(buf1, 63, "%Y-%m-%d_%H-%M-%S", ptm);
+                len = strlen(buf1);
+                if ((q - buf + len) > buf_size - 1)
+                    goto fail;
+                memcpy(q, buf1, len);
+                q += len;
+                break;
             default:
                 goto fail;
             }
@@ -4135,7 +4155,7 @@
                 *q++ = c;
         }
     }
-    if (!percentd_found)
+    if (!percentd_found && !percentt_found)
         goto fail;
     *q = '\0';
     return 0;
