--- ffmpeg-1.0/libavformat/movenc.c	2012-09-28 03:37:35.000000000 +0200
@@ -1552,7 +1552,11 @@
     (version == 1) ? avio_wb32(pb, 104) : avio_wb32(pb, 92); /* size */
     ffio_wfourcc(pb, "tkhd");
     avio_w8(pb, version);
+    if (st->index < 2) {
     avio_wb24(pb, 0xf); /* flags (track enabled) */
+    } else {
+        avio_wb24(pb, 0xe); /* flags (track present but not enabled) */
+    }
     if (version == 1) {
         avio_wb64(pb, track->time);
         avio_wb64(pb, track->time);
