--- Makefile.PL.orginal	2008-09-05 15:36:24.000000000 +0000
+++ Makefile.PL	2009-02-08 14:31:56.166087046 +0000
@@ -21,20 +21,22 @@
 build_requires   'LWP::UserAgent'               =>    0;
 build_requires   'LWP::ConnCache'               =>    0;
 
-my $value = prompt( 'Do you want to test streaming ( requires POE::Component::Client::HTTP ) [y/N]?', 'N' );
-build_requires   'POE::Component::Client::HTTP' => 0.82 if $value =~ /^Y$/i;
-
-# Ask users if they want SSL support
-$value = prompt( 'Do you want SSL support ( requires POE::Component::SSLify ) [y/N]?', 'N' );
+eval { require POE::Component::Client::HTTP; };
+unless($@) {
+    print "with POE::Component::Client::HTTP\n";
+    build_requires   'POE::Component::Client::HTTP' => 0.82;
+}
 
 # Add to the prereqs PoCo::SSLify?
-requires 'POE::Component::SSLify' => '0.04' if $value =~ /^Y$/i;
-
-# Ask users if they want PreFork support
-$value = prompt( 'Do you want PreFork support ( requires IPC::Shareable ) [y/N]?', 'N' );
-
+eval { require POE::Component::SSLify; };
+unless($@) {
+    print "with POE::Component::SSLify\n";
+    requires 'POE::Component::SSLify' => '0.04';
+}
 # Add to the prereqs IPC::Shareable?
-if ( $value =~ /^Y$/i ) {
+eval { require IPC::Shareable; };
+unless($@) {
+    print "with Prefork IPC::Shareable\n";
    requires 'IPC::Shareable' => '0';
    build_requires   'POE::Component::Client::HTTP' => 0.82;
 }
