--- old/policyd-weight	2016-04-18 10:59:26.987884246 +0200
+++ new/policyd-weight	2016-04-18 10:35:32.699852644 +0200
@@ -69,7 +69,6 @@
 use Sys::Syslog qw(:DEFAULT setlogsock);
 use Net::DNS;
 use Net::IP;
-use Net::DNS::Packet qw(dn_expand);
 use IO::Socket::INET;
 use IO::Socket::UNIX;
 use IO::Select;
@@ -3588,7 +3587,8 @@
             my    ($id, $bf, $qc, $anc, $nsc, $arc, $qb) = 
             unpack('n   n    n    n     n     n     a*', $buf);
 
-            my ($dn, $offset) = dn_expand(\$qb, 0);
+	    my ($decoded, $offset) = decode Net::DNS::DomainName(\$qb);
+	    my $dn = $decoded->name;
 
             if(($id && $anc) && ($id == $oid) && ($query eq $dn))
             {
