Source: https://bugzilla.redhat.com/show_bug.cgi?id=801650

--- sysdeps/x86_64/multiarch/init-arch.c	2012-05-10 14:06:54.607713791 -0600
+++ sysdeps/x86_64/multiarch/init-arch.c	2012-05-10 14:15:37.855008615 -0600
@@ -154,6 +154,8 @@ __init_cpu_features (void)
 		     : "=a" (xcrlow), "=d" (xcrhigh) : "c" (0));
 		(xcrlow & 6) == 6; }))
 	__cpu_features.feature[index_YMM_Usable] |= bit_YMM_Usable;
+      else
+        __cpu_features.cpuid[COMMON_CPUID_INDEX_1].ecx &= ~bit_AVX;
     }
 
   __cpu_features.family = family;
