--- glusterfsd.orig	2009-06-22 16:28:17.000000000 +0000
+++ glusterfsd	2009-06-16 02:04:31.000000000 +0000
@@ -2,6 +2,14 @@
 # Copyright 1999-2009 Gentoo Foundation
 # Distributed under the terms of the GNU General Public License v2
 
+VOLDIR="/etc/glusterfs"
+VOL="${SVCNAME#*.}"
+if [ -n "${VOL}" ] && [ "${SVCNAME}" != "glusterfs" ]; then
+        VOLPID="/var/run/glusterfs.${VOL}.pid"
+else
+        VOLPID=/var/run/glusterfs.pid
+fi
+
 depend() {
 	use net dns
 }
@@ -9,15 +17,15 @@
 start() {
 	ebegin "Starting ${SVCNAME}"
 	start-stop-daemon --start --exec /usr/sbin/glusterfsd -- \
-		--volfile=/etc/glusterfs/${SVCNAME}.vol \
-		--pid-file=/var/run/${SVCNAME}.pid \
-		--log-file=/var/log/glusterfs/${SVCNAME}.log
+		--volfile=/etc/glusterfs/${VOL}.vol \
+		--pid-file=${VOLPID} \
+		--log-file=/var/log/glusterfs/${VOL}.log \
+		${GLUSTERFSD_MOUNTPOINT}
 	eend $?
 }
 
 stop() {
 	ebegin "Stopping ${SVCNAME}"
-	start-stop-daemon --stop --pidfile /var/run/${SVCNAME}.pid
+	start-stop-daemon --stop --pidfile ${VOLPID}
 	eend $?
 }
-
