diff -ru /usr/share/php/PHP/Depend/Builder/Default.php PHP/Depend/Builder/Default.php
--- php/PHP/Depend/Builder/Default.php	2012-08-26 22:01:21.000000000 +0200
+++ PHP_Depend-1.0.7/PHP/Depend/Builder/Default.php	2012-05-23 02:15:51.000000000 +0200
@@ -514,7 +514,7 @@
      * @since 0.9.6
      */
     public function buildASTSelfReference(
-        PHP_Depend_Code_AbstractClassOrInterface $type
+        PHP_Depend_Code_AbstractType $type
     ) {
         PHP_Depend_Util_Log::debug(
             'Creating: PHP_Depend_Code_ASTSelfReference(' . $type->getName() . ')'
@@ -554,7 +554,7 @@
      * @since 0.9.6
      */
     public function buildASTStaticReference(
-        PHP_Depend_Code_AbstractClassOrInterface $owner
+        PHP_Depend_Code_AbstractType $owner
     ) {
         PHP_Depend_Util_Log::debug('Creating: PHP_Depend_Code_ASTStaticReference()');
 
diff -ru /usr/share/php/PHP/Depend/BuilderI.php PHP/Depend/BuilderI.php
--- php/PHP/Depend/BuilderI.php	2012-08-26 22:01:22.000000000 +0200
+++ PHP_Depend-1.0.7/PHP/Depend/BuilderI.php	2012-05-23 02:16:02.000000000 +0200
@@ -241,7 +241,7 @@
      * @return PHP_Depend_Code_ASTSelfReference
      * @since 0.9.6
      */
-    function buildASTSelfReference(PHP_Depend_Code_AbstractClassOrInterface $type);
+    function buildASTSelfReference(PHP_Depend_Code_AbstractType $type);
 
     /**
      * Builds a new parent reference instance.
@@ -266,7 +266,7 @@
      * @since 0.9.6
      */
     function buildASTStaticReference(
-        PHP_Depend_Code_AbstractClassOrInterface $owner
+        PHP_Depend_Code_AbstractType $owner
     );
 
     /**
diff -ru /usr/share/php/PHP/Depend/Code/ASTSelfReference.php PHP/Depend/Code/ASTSelfReference.php
--- php/PHP/Depend/Code/ASTSelfReference.php	2012-08-26 22:01:21.000000000 +0200
+++ PHP_Depend-1.0.7/PHP/Depend/Code/ASTSelfReference.php	2012-05-23 02:13:17.000000000 +0200
@@ -101,7 +101,7 @@
      */
     public function __construct(
         PHP_Depend_Builder_Context $context,
-        PHP_Depend_Code_AbstractClassOrInterface $target
+        PHP_Depend_Code_AbstractType $target
     ) {
         $this->context      = $context;
         $this->typeInstance = $target;
diff -ru package.xml.orig package.xml
--- package.xml.orig	2012-08-26 22:45:30.911788911 +0200
+++ package.xml	2012-08-26 22:46:44.377798252 +0200
@@ -164,7 +164,7 @@
     <tasks:replace from="@bin_dir@" to="bin_dir" type="pear-config" />
     <tasks:replace from="/usr/bin/env php" to="php_bin" type="pear-config" />
    </file>
-   <file md5sum="acce5c4e8fb1a89bccc2d27fd3618b88" name="PHP/Depend/Builder/Default.php" role="php">
+   <file md5sum="c7de8f4a0d64dd8cdaa7ec6fc813a72b" name="PHP/Depend/Builder/Default.php" role="php">
     <tasks:replace from="@package_version@" to="version" type="package-info" />
     <tasks:replace from="@data_dir@" to="data_dir" type="pear-config" />
     <tasks:replace from="@php_bin@" to="php_bin" type="pear-config" />
@@ -633,7 +633,7 @@
     <tasks:replace from="@bin_dir@" to="bin_dir" type="pear-config" />
     <tasks:replace from="/usr/bin/env php" to="php_bin" type="pear-config" />
    </file>
-   <file md5sum="04a816f4194e1c4d975bf996f931554c" name="PHP/Depend/Code/ASTSelfReference.php" role="php">
+   <file md5sum="85649105a9aadb39cc923696a5488523" name="PHP/Depend/Code/ASTSelfReference.php" role="php">
     <tasks:replace from="@package_version@" to="version" type="package-info" />
     <tasks:replace from="@data_dir@" to="data_dir" type="pear-config" />
     <tasks:replace from="@php_bin@" to="php_bin" type="pear-config" />
@@ -1578,7 +1578,7 @@
     <tasks:replace from="@bin_dir@" to="bin_dir" type="pear-config" />
     <tasks:replace from="/usr/bin/env php" to="php_bin" type="pear-config" />
    </file>
-   <file md5sum="dceb71f194917ea6511c6aa5590eeb7d" name="PHP/Depend/BuilderI.php" role="php">
+   <file md5sum="a9ee72b409ee833d6b002b6335fc21bf" name="PHP/Depend/BuilderI.php" role="php">
     <tasks:replace from="@package_version@" to="version" type="package-info" />
     <tasks:replace from="@data_dir@" to="data_dir" type="pear-config" />
     <tasks:replace from="@php_bin@" to="php_bin" type="pear-config" />
