From d1d6f87174dd0e733539566a4bbf0599e27adc5b Mon Sep 17 00:00:00 2001
From: Jan Engelhardt <jengelh@medozas.de>
Date: Fri, 30 Mar 2012 03:51:13 +0200
Subject: [PATCH] build: do not compile pmt-ehd if no good libcryptsetup is found

---
 src/Makefile.am |    4 +++-
 1 files changed, 3 insertions(+), 1 deletions(-)

diff --git a/src/Makefile.am b/src/Makefile.am
index 74e1776..6447b42 100644
--- a/src/Makefile.am
+++ b/src/Makefile.am
@@ -8,7 +8,9 @@ AM_CFLAGS = ${regular_CFLAGS} ${GCC_FVISIBILITY_HIDDEN} \
 moduledir		= @PAM_MODDIR@
 module_LTLIBRARIES	= pam_mount.la
 sbin_PROGRAMS		= pmvarrun
+if HAVE_LIBCRYPTSETUP
 sbin_PROGRAMS		+= pmt-ehd
+endif
 ssbin_PROGRAMS		= mount.crypt
 noinst_PROGRAMS		= autoloop
 noinst_SCRIPTS 		= umount.crypt
@@ -71,7 +73,7 @@ mount_crypt_SOURCES	= mtcrypt.c spawn.c
 mount_crypt_LDADD	= libcryptmount.la libpmt_mtab.la ${libHX_LIBS}
 
 pmt_ehd_SOURCES		= ehd.c bdev.c misc.c spawn.c
-pmt_ehd_LDADD		= libcryptmount.la ${libHX_LIBS}
+pmt_ehd_LDADD		= libcryptmount.la ${libHX_LIBS} ${libcryptsetup_LIBS}
 
 #
 # runtime helpers
-- 
1.7.4.1

