--- qemu-xen-3.4-testing.git/xen-setup.orig	2009-11-03 11:28:55.000000000 +0100
+++ qemu-xen-3.4-testing.git/xen-setup	2009-11-03 11:29:07.000000000 +0100
@@ -3,6 +3,8 @@
 
 # git-clean -x -d && ./xen-setup && make prefix=/usr CMDLINE_CFLAGS='-O0 -g' -j4 && make install DESTDIR=`pwd`/dist/ prefix=/usr && rsync -a --stats --delete . thule:shadow/qemu-iwj.git/ && rsync -a --stats dist/. root@thule:/
 
+target=i386-dm
+
 rm -f $target/Makefile
 rm -f $target/config.mak
 rm -f config-host.mak
@@ -11,8 +13,6 @@
 
 ./configure --disable-gfx-check --disable-curses --disable-slirp "$@" --prefix=/usr
 
-target=i386-dm
-
 if [ "x$XEN_ROOT" != x ]; then
 	echo "XEN_ROOT=$XEN_ROOT" >>config-host.mak
 fi
