--- tools/hotplug/Linux/Makefile        2010-07-20 10:42:39.000000000 +0200
+++ tools/hotplug/Linux/Makefile        2010-07-20 10:42:56.000000000 +0200
@@ -10,6 +10,7 @@
 # Xen script dir and scripts to go there.
 XEN_SCRIPTS = network-bridge vif-bridge
 XEN_SCRIPTS += network-route vif-route
+XEN_SCRIPTS += network-route6 vif-route6
 XEN_SCRIPTS += network-nat vif-nat
 XEN_SCRIPTS += vif2
 XEN_SCRIPTS += block
--- tools/hotplug/Linux/network-route6	2010-07-20 10:04:28.000000000 +0200
+++ tools/hotplug/Linux/network-route6	2010-07-20 09:49:23.000000000 +0200
@@ -0,0 +1,31 @@
+#!/bin/bash
+#============================================================================
+# Default Xen network start/stop script.
+# Xend calls a network script when it starts.
+# The script name to use is defined in ${XEN_CONFIG_DIR}/xend-config.sxp
+# in the network-script field.
+#
+# Usage:
+#
+# network-route (start|stop|status) {VAR=VAL}*
+#
+# Vars:
+#
+# netdev     The gateway interface (default eth0).
+# antispoof  Whether to use iptables to prevent spoofing (default yes).
+#
+#============================================================================
+
+dir=$(dirname "$0")
+. "$dir/hotplugpath.sh"
+. "$dir/xen-script-common.sh"
+
+evalVariables "$@"
+
+netdev=${netdev:-eth${vifnum}}
+
+echo 1 >/proc/sys/net/ipv4/ip_forward
+echo 1 >/proc/sys/net/ipv4/conf/eth0/proxy_arp
+
+echo 1 >/proc/sys/net/ipv6/conf/all/forwarding
+echo 1 >/proc/sys/net/ipv6/conf/all/proxy_ndp
--- tools/hotplug/Linux/vif-common.sh	2010-07-19 13:11:09.000000000 +0200
+++ tools/hotplug/Linux/vif-common.sh	2010-07-19 13:43:43.000000000 +0200
@@ -84,6 +84,22 @@
   fi
 }
 
+frob_ip6table()
+{
+  if [ "$command" == "online" ]
+  then
+    local c="-A"
+  else
+    local c="-D"
+  fi
+  
+  log info "[vif-route] running ip6tables $c FORWARD $vif $@"
+  ip6tables "$c" FORWARD -m physdev --physdev-in "$vif" "$@" -j ACCEPT \
+    2>/dev/null ||
+    [ "$c" == "-D" ] ||
+    log err \
+    "ip6tables $c FORWARD -m physdev --physdev-in $vif $@ -j ACCEPT failed. If you are using iptables, this may affect networking for guest domains."
+}
 
 ##
 # Add or remove the appropriate entries in the iptables.  With antispoofing
@@ -104,13 +120,19 @@
   fi
 
   claim_lock "iptables"
+  
+  log info "[vif-route] Adding iptables FORWARD $vif $@"
 
   if [ "$ip" != "" ]
   then
       local addr
       for addr in $ip
       do
-        frob_iptable -s "$addr"
+        if [[ $addr =~ ^[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}$ ]]; then
+          frob_iptable -s "$addr"
+        else
+          frob_ip6table -s "$addr"
+        fi
       done
 
       # Always allow the domain to talk to a DHCP server.
@@ -135,6 +157,10 @@
   ip addr show "$1" | awk "/^.*inet.*$1\$/{print \$2}" | sed -n '1 s,/.*,,p'
 }
 
+ip6_of() {
+  log info "[vif-route] found dom0_netdev $1"
+  ip -6 addr show "$1" | perl -wane '/scope global/ && /inet6 (([0-9a-f]+:*)+)/ && print $1;'
+}
 
 ##
 # dom0_ip
@@ -156,3 +182,15 @@
   fi
   echo "$result"
 }
+
+dom0_ip6() {
+  local nd6=${netdev:-eth0}
+  local result=$(ip6_of "$nd6")
+  if [ -z "$result" ]
+  then
+    fatal
+"$netdev is not up.  Bring it up or specify another interface with " \
+"netdev=<if> as a parameter to $0."
+  fi
+  echo "$result"
+}
--- tools/hotplug/Linux/vif-route6	2010-07-20 10:04:21.000000000 +0200
+++ tools/hotplug/Linux/vif-route6	2010-07-20 10:01:21.000000000 +0200
@@ -0,0 +1,78 @@
+#!/bin/bash
+#============================================================================
+# ${XEN_SCRIPT_DIR}/vif-route6
+#
+# Script for configuring a vif in routed mode.
+# The hotplugging system will call this script if it is specified either in
+# the device configuration given to Xend, or the default Xend configuration
+# in ${XEN_CONFIG_DIR}/xend-config.sxp.  If the script is specified in
+# neither of those places, then vif-bridge is the default.
+#
+# Usage:
+# vif-route (add|remove|online|offline)
+#
+# Environment vars:
+# vif         vif interface name (required).
+# XENBUS_PATH path to this device's details in the XenStore (required).
+#
+# Read from the store:
+# ip      list of IP networks for the vif, space-separated (default given in
+#         this script).
+#============================================================================
+
+dir=$(dirname "$0")
+. "$dir/vif-common.sh"
+
+main_ip=$(dom0_ip)
+main_ip6=$(dom0_ip6)
+
+case "$command" in
+    online)
+        log info "[vif-route] online request, ip ${ip} with main_ip ${main_ip} and main_ip6 ${main_ip6} for $vif."
+        ifconfig ${vif} ${main_ip} netmask 255.255.255.255 up
+        echo 1 >/proc/sys/net/ipv4/conf/${vif}/proxy_arp
+        ipcmd='add'
+        cmdprefix=''
+        if [ -n "${main_ip6}" ] ; then
+          log info "[vif-route6] enabling proxy_ndp for interface ${vif}"
+          echo 1 > /proc/sys/net/ipv6/conf/${vif}/proxy_ndp
+          log info "[vif-route6] adding ${main_ip6} to ${vif}"
+          ip -6 addr add ${main_ip6} dev ${vif}
+        fi
+        ;;
+    offline)
+        do_without_error ifdown ${vif}
+        ipcmd='del'
+        cmdprefix='do_without_error'
+        ;;
+esac
+
+if [ "${ip}" ] ; then
+    # If we've been given a list of IP addresses, then add routes from dom0 to
+    # the guest using those addresses.
+    for addr in ${ip} ; do
+      if [[ $addr =~ ^[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}$ ]]; then
+        log info "[vif-route6] ${ipcmd} IPv4 address ${addr} with src ${main_ip} for $vif."
+        ${cmdprefix} ip route ${ipcmd} ${addr} dev ${vif} src ${main_ip}
+      else
+        log info "[vif-route6] ${ipcmd} IPv6 address ${addr} with src ${main_ip6} for $vif."
+        ${cmdprefix} ip -6 route ${ipcmd} ${addr} dev ${vif}
+        log info "[vif-route6] ${ipcmd} proxy_ndp for address ${addr%%/*}"
+        log info "ip -6 neigh ${ipcmd} proxy ${addr%%/*} dev ${netdev:-eth0}"
+        ${cmdprefix} ip -6 neigh ${ipcmd} proxy ${addr%%/*} dev ${netdev:-eth0}
+      fi
+    done 
+fi
+
+# add/remove vif to internalbridge
+if [ `echo "${vif}" | grep -e ".*\-br$"` ]; then
+    brctl ${ipcmd}if xenbr0 ${vif}
+fi
+
+handle_iptable
+
+log debug "Successful vif-route $command for $vif."
+if [ "$command" = "online" ]
+then
+  success
+fi
