--- tools/examples/xend-config.sxp.orig	2012-02-28 03:56:59.020529912 +0100
+++ tools/examples/xend-config.sxp	2012-02-28 03:57:14.461252017 +0100
@@ -170,7 +170,7 @@
 # two fake interfaces per guest domain.  To do things like this, write
 # yourself a wrapper script, and call network-bridge from it, as appropriate.
 #
-(network-script network-bridge)
+(network-script /bin/true)
 
 # The script used to control virtual interfaces.  This can be overridden on a
 # per-vif basis when creating a domain or a configuring a new vif.  The
