--- Translation/TrlUtil/MainForm.cs
+++ Translation/TrlUtil/MainForm.cs
@@ -1,4 +1,4 @@
-/*
+ï»¿/*
   KeePass Password Safe - The Open-Source Password Manager
   Copyright (C) 2003-2013 Dominik Reichl <dominik.reichl@t-online.de>
 
@@ -594,8 +594,8 @@
 					string strTrl = kpi.Value;
 					if(string.IsNullOrEmpty(strEn) || string.IsNullOrEmpty(strTrl)) continue;
 
-					bool bEllEn = (strEn.EndsWith(@"...") || strEn.EndsWith(@"…"));
-					bool bEllTrl = (strTrl.EndsWith(@"...") || strTrl.EndsWith(@"…"));
+					bool bEllEn = (strEn.EndsWith(@"...") || strEn.EndsWith(@"Â…"));
+					bool bEllTrl = (strTrl.EndsWith(@"...") || strTrl.EndsWith(@"Â…"));
 
 					if(bEllEn && !bEllTrl)
 						MessageService.ShowWarning("Warning! The English string",
@@ -926,10 +926,10 @@
 
 				vNodes.Add(tn);
 				if(kpfc != null)
-					vValues.Add(kpfc.Window.Name + "©" + kpfc.Window.TextEnglish +
-						"©" + kpfc.Window.Text);
+					vValues.Add(kpfc.Window.Name + "Â©" + kpfc.Window.TextEnglish +
+						"Â©" + kpfc.Window.Text);
 				else if(kpcc != null)
-					vValues.Add(kpcc.Name + "©" + kpcc.TextEnglish + "©" + kpcc.Text);
+					vValues.Add(kpcc.Name + "Â©" + kpcc.TextEnglish + "Â©" + kpcc.Text);
 				else vValues.Add(tn.Text);
 
 				GetControlTreeItems(tn.Nodes, vNodes, vValues);
--- KeePassLib/Cryptography/QualityEstimation.cs
+++ KeePassLib/Cryptography/QualityEstimation.cs
@@ -1,4 +1,4 @@
-/*
+ï»¿/*
   KeePass Password Safe - The Open-Source Password Manager
   Copyright (C) 2003-2013 Dominik Reichl <dominik.reichl@t-online.de>
 
@@ -592,7 +592,7 @@
 				case '\u00D1':
 				case '\u00F1': ch = 'n'; break;
 				case '0':
-				case '°':
+				case 'Â°':
 				case '*':
 				case '\u00A4':
 				case '\u00D8':
--- KeePassLib/PwDefs.cs
+++ KeePassLib/PwDefs.cs
@@ -1,4 +1,4 @@
-/*
+ï»¿/*
   KeePass Password Safe - The Open-Source Password Manager
   Copyright (C) 2003-2013 Dominik Reichl <dominik.reichl@t-online.de>
 
@@ -68,7 +68,7 @@
 		/// </summary>
 		public const string VersionString = "2.23";
 
-		public const string Copyright = @"Copyright © 2003-2013 Dominik Reichl";
+		public const string Copyright = @"Copyright Â© 2003-2013 Dominik Reichl";
 
 		/// <summary>
 		/// Product website URL. Terminated by a forward slash.
