--- desktopvideo-9.0a15-x86_64/usr/src/desktopvideo-9.0a15/blackmagic_core.c.orig	2011-11-23 00:49:12.000000000 +0100
+++ desktopvideo-9.0a15-x86_64/usr/src/desktopvideo-9.0a15/blackmagic_core.c	2011-11-23 00:49:45.000000000 +0100
@@ -327,9 +327,9 @@
 	pci_enable_device(pdev);
 	pci_set_master(pdev);
 
-	if (pci_set_dma_mask(pdev, DMA_64BIT_MASK) < 0)
+	if (pci_set_dma_mask(pdev, DMA_BIT_MASK(64)) < 0)
 	{
-		if (pci_set_dma_mask(pdev, DMA_32BIT_MASK) < 0)
+		if (pci_set_dma_mask(pdev, DMA_BIT_MASK(32)) < 0)
 			goto fail;
 	}
 	
