https://bugzilla.redhat.com/show_bug.cgi?id=782369
---

--- src/modules/perl/modperl_config.c
+++ src/modules/perl/modperl_config.c
@@ -186,7 +186,7 @@
 
     /* make sure httpd's argv[0] is the first argument so $0 is
      * correctly connected to the real thing */
-    modperl_config_srv_argv_push(s->process->argv[0]);
+    modperl_config_srv_argv_push(s->process->short_name);
 
     MP_TRACE_d(MP_FUNC, "new scfg: 0x%lx", (unsigned long)scfg);
