--- lib/RT/Client/REST/Forms.pm.old	2008-12-17 02:09:04.000000000 +0100
+++ lib/RT/Client/REST/Forms.pm	2008-12-17 02:10:29.000000000 +0100
@@ -15,7 +15,7 @@
 @ISA = qw(Exporter);
 @EXPORT = qw(expand_list form_parse form_compose vpush vsplit);
 
-my $field = '(?:[a-zA-Z][a-zA-Z0-9_-]*|CF-[a-zA-Z0-9_ -]*)';
+my $field = '(?:[a-zA-Z][a-zA-Z0-9_-]*|CF.{[a-zA-Z0-9_ -]*})';
 
 sub expand_list {
     my ($list) = @_;
--- lib/RT/Client/REST/Object.pm.old	2008-12-17 02:11:00.000000000 +0100
+++ lib/RT/Client/REST/Object.pm	2008-12-17 02:11:50.000000000 +0100
@@ -395,7 +395,7 @@
     }
     my @cfs = ($all ? $self->cf : $self->_dirty_cf);
     for my $cf (@cfs) {
-        $hash{'CF-' . $cf} = $self->cf($cf);
+        $hash{'CF.{' . $cf . '}'} = $self->cf($cf);
     }
 
     return \%hash;
