--- configure.~1~	2012-01-04 10:04:44.000000000 +0100
+++ configure	2012-01-04 10:08:17.376023243 +0100
@@ -5003,10 +5003,13 @@
 
 # Check whether --enable-command-args was given.
 if test "${enable_command_args+set}" = set; then :
-  enableval=$enable_command_args; cat >>confdefs.h <<_ACEOF
+  enableval=$enable_command_args;
+  if test x$enableval = xyes; then
+    cat >>confdefs.h <<_ACEOF
 #define ENABLE_COMMAND_ARGUMENTS 1
 _ACEOF
 
+  fi
 fi
 
 
