--- src/check_nrpe.old	2012-01-04 08:38:12.000000000 +0100
+++ src/check_nrpe.c	2012-01-04 08:38:42.000000000 +0100
@@ -416,6 +416,7 @@
 		case 'H':
 			server_name = strdup(optarg);
 			break;
+#ifdef HAVE_SSL
 		case 'C':
 			cert_file = strdup(optarg);
 			break;
@@ -425,6 +426,7 @@
 		case 'k':
 			privatekey_file = strdup(optarg);
 			break;
+#endif
 		case 'c':
 			command_name = strdup(optarg);
 			break;
@@ -466,8 +468,10 @@
 	if (server_name == NULL && show_help == FALSE && show_version == FALSE  && show_license == FALSE)
 		return ERROR;
 
+#ifdef HAVE_SSL
 	if ((cert_file == NULL || cacert_file == NULL || privatekey_file == NULL) && use_ssl && show_help == FALSE && show_version == FALSE && show_license == FALSE)
 		return ERROR;
+#endif
 
 	return OK;
 }
--- src/nrpe.old	2012-01-04 08:38:20.000000000 +0100
+++ src/nrpe.c	2012-01-04 08:39:12.000000000 +0100
@@ -99,9 +99,7 @@
 	int x;
 	char buffer[MAX_INPUT_BUFFER];
 	char *env_string = NULL;
-#ifdef HAVE_SSL
 	int nullfd;
-#endif
 
 	/* set some environment variables */
 	asprintf(&env_string, "NRPE_MULTILINESUPPORT=1");
