--- galera-23.2.2/galera/src/saved_state.cpp	2012-10-26 13:48:39.000000000 +0200
+++ galera-23.2.2-src/galera/src/saved_state.cpp	2012-12-29 22:49:31.816495710 +0100
@@ -235,7 +235,8 @@
             buf[write_size] = ' '; // overwrite whatever is there currently
 
         rewind(fs_);
-        fwrite(buf, write_size, 1, fs_);
+        size_t wsize;
+        wsize = fwrite(buf, write_size, 1, fs_);
         fflush(fs_);
 
         current_len_ = state_len;
