--- /var/tmp/portage/dev-db/galera-23.2.2/work/galera-23.2.2/SConstruct	2012-10-26 13:48:39.000000000 +0200
+++ /usr/src/23.2.x/SConstruct	2013-01-01 18:16:01.586553548 +0100
@@ -98,7 +98,7 @@
 boost = int(ARGUMENTS.get('boost', 1))
 ssl   = int(ARGUMENTS.get('ssl', 1))
 
-GALERA_VER = ARGUMENTS.get('version', '2.2')
+GALERA_VER = ARGUMENTS.get('version', '2.3dev')
 GALERA_REV = ARGUMENTS.get('revno', 'XXXX')
 # export to any module that might have use of those
 Export('GALERA_VER', 'GALERA_REV')
@@ -239,6 +239,10 @@
     Exit(1)
 conf.env.Append(CPPFLAGS = ' -DHAVE_BOOST_SHARED_PTR_HPP')
 
+if conf.CheckCXXHeader('boost/date_time/posix_time/posix_time_types.hpp'):
+    conf.env.Append(CPPFLAGS = ' -DBOOST_DATE_TIME_POSIX_TIME_STD_CONFIG -lboost_date_time-mt -lboost_thread-mt')
+    env.Append(LINKFLAGS = ' -lboost_system')
+
 if conf.CheckCXXHeader('tr1/unordered_map'):
     conf.env.Append(CPPFLAGS = ' -DHAVE_TR1_UNORDERED_MAP')
 else:
