diff -ruN /usr/share/php/PHP/PMD/Node/AbstractType.php PHP/PMD/Node/AbstractType.php
--- php/PHP/PMD/Node/AbstractType.php	2012-08-26 22:01:34.000000000 +0200
+++ PHP_PMD-1.3.3/PHP/PMD/Node/AbstractType.php	2012-05-23 02:56:42.000000000 +0200
@@ -67,7 +67,7 @@
      *
      * @param PHP_Depend_Code_AbstractClassOrInterface $node The wrapped node.
      */
-    public function __construct(PHP_Depend_Code_AbstractClassOrInterface $node)
+    public function __construct(PHP_Depend_Code_AbstractType $node)
     {
         parent::__construct($node);
     }
diff -ruN /usr/share/php/PHP/PMD/Node/Method.php PHP/PMD/Node/Method.php
--- php/PHP/PMD/Node/Method.php	2012-08-26 22:01:34.000000000 +0200
+++ PHP_PMD-1.3.3/PHP/PMD/Node/Method.php	2012-05-23 03:11:19.000000000 +0200
@@ -131,6 +131,9 @@
         if ($parentNode instanceof PHP_Depend_Code_Class) {
             return new PHP_PMD_Node_Class($parentNode);
         }
+        if ($parentNode instanceof PHP_Depend_Code_Trait) {
+            return new PHP_PMD_Node_Trait($parentNode);
+        }
         return new PHP_PMD_Node_Interface($parentNode);
     }
 
@@ -150,6 +153,10 @@
         $methodName = strtolower($this->getName());
 
         $parentNode = $this->getNode()->getParent();
+        if($parentNode instanceof PHP_Depend_Code_Trait) {
+            return true;
+        }
+
         foreach ($parentNode->getInterfaces() as $parentType) {
             $methods = $parentType->getAllMethods();
             if (isset($methods[$methodName])) {
diff -ruN /usr/share/php/PHP/PMD/Node/Trait.php PHP/PMD/Node/Trait.php
--- php/PHP/PMD/Node/Trait.php	1970-01-01 01:00:00.000000000 +0100
+++ PHP_PMD-1.3.3/PHP/PMD/Node/Trait.php	2012-08-26 22:26:25.320787424 +0200
@@ -0,0 +1,79 @@
+<?php
+/**
+ * This file is part of PHP_PMD.
+ *
+ * PHP Version 5
+ *
+ * Copyright (c) 2008-2012, Manuel Pichler <mapi@phpmd.org>.
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ *
+ *   * Redistributions of source code must retain the above copyright
+ *     notice, this list of conditions and the following disclaimer.
+ *
+ *   * Redistributions in binary form must reproduce the above copyright
+ *     notice, this list of conditions and the following disclaimer in
+ *     the documentation and/or other materials provided with the
+ *     distribution.
+ *
+ *   * Neither the name of Manuel Pichler nor the names of his
+ *     contributors may be used to endorse or promote products derived
+ *     from this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+ * FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ * COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ * BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ * LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ * ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ *
+ * @category   PHP
+ * @package    PHP_PMD
+ * @subpackage Node
+ * @author     Manuel Pichler <mapi@phpmd.org>
+ * @copyright  2008-2012 Manuel Pichler. All rights reserved.
+ * @license    http://www.opensource.org/licenses/bsd-license.php  BSD License
+ * @version    SVN: $Id$
+ * @link       http://phpmd.org
+ */
+
+require_once 'PHP/PMD/Node/AbstractType.php';
+
+/**
+ * Wrapper around PHP_Depend's Trait objects.
+ *
+ * @category   PHP
+ * @package    PHP_PMD
+ * @subpackage Node
+ * @author     Manuel Pichler <mapi@phpmd.org>
+ * @copyright  2008-2012 Manuel Pichler. All rights reserved.
+ * @license    http://www.opensource.org/licenses/bsd-license.php  BSD License
+ * @version    Release: 1.3.3
+ * @link       http://phpmd.org
+ */
+class PHP_PMD_Node_Trait extends PHP_PMD_Node_AbstractType
+{
+    /**
+     * The type of this Trait.
+     */
+    const CLAZZ = __CLASS__;
+
+    /**
+     * Constructs a new Trait wrapper node.
+     *
+     * @param PHP_Depend_Code_Trait $node The wrapped Trait object.
+     */
+    public function __construct(PHP_Depend_Code_Trait $node)
+    {
+        parent::__construct($node);
+    }
+}
diff -ruN /usr/share/php/PHP/PMD/Parser.php PHP/PMD/Parser.php
--- php/PHP/PMD/Parser.php	2012-08-26 22:01:34.000000000 +0200
+++ PHP_PMD-1.3.3/PHP/PMD/Parser.php	2012-05-23 02:55:06.000000000 +0200
@@ -50,6 +50,7 @@
 require_once 'PHP/PMD/Node/Function.php';
 require_once 'PHP/PMD/Node/Interface.php';
 require_once 'PHP/PMD/Node/Method.php';
+require_once 'PHP/PMD/Node/Trait.php';
 
 /**
  * Simple wrapper around the php depend engine.
--- package.xml.orig	2012-08-26 22:33:00.441787942 +0200
+++ package.xml	2012-08-26 22:34:34.046812108 +0200
@@ -106,7 +106,14 @@
     <tasks:replace from="@bin_dir@" to="bin_dir" type="pear-config" />
     <tasks:replace from="/usr/bin/env php" to="php_bin" type="pear-config" />
    </file>
-   <file md5sum="b3f9a82d140053425f67c930ce8fe17e" name="PHP/PMD/Node/Method.php" role="php">
+   <file md5sum="ea1ca205ad1ce0a36b57b5b630c59ea8" name="PHP/PMD/Node/Trait.php" role="php">
+    <tasks:replace from="@package_version@" to="version" type="package-info" />
+    <tasks:replace from="@data_dir@" to="data_dir" type="pear-config" />
+    <tasks:replace from="@php_bin@" to="php_bin" type="pear-config" />
+    <tasks:replace from="@bin_dir@" to="bin_dir" type="pear-config" />
+    <tasks:replace from="/usr/bin/env php" to="php_bin" type="pear-config" />
+   </file>
+   <file md5sum="0c9286b69c742872b0a32af48cc8cbd0" name="PHP/PMD/Node/Method.php" role="php">
     <tasks:replace from="@package_version@" to="version" type="package-info" />
     <tasks:replace from="@data_dir@" to="data_dir" type="pear-config" />
     <tasks:replace from="@php_bin@" to="php_bin" type="pear-config" />
@@ -120,7 +127,7 @@
     <tasks:replace from="@bin_dir@" to="bin_dir" type="pear-config" />
     <tasks:replace from="/usr/bin/env php" to="php_bin" type="pear-config" />
    </file>
-   <file md5sum="4c42be79fedbaf4fec414cae7610ba2a" name="PHP/PMD/Node/AbstractType.php" role="php">
+   <file md5sum="1c16a238e23054a65dc833eb923dacd0" name="PHP/PMD/Node/AbstractType.php" role="php">
     <tasks:replace from="@package_version@" to="version" type="package-info" />
     <tasks:replace from="@data_dir@" to="data_dir" type="pear-config" />
     <tasks:replace from="@php_bin@" to="php_bin" type="pear-config" />
@@ -449,7 +456,7 @@
     <tasks:replace from="@bin_dir@" to="bin_dir" type="pear-config" />
     <tasks:replace from="/usr/bin/env php" to="php_bin" type="pear-config" />
    </file>
-   <file md5sum="92384de16c002e6fda7def621f8126ea" name="PHP/PMD/Parser.php" role="php">
+   <file md5sum="08ebd68b7a5887099d3235dbdd6bfd8b" name="PHP/PMD/Parser.php" role="php">
     <tasks:replace from="@package_version@" to="version" type="package-info" />
     <tasks:replace from="@data_dir@" to="data_dir" type="pear-config" />
     <tasks:replace from="@php_bin@" to="php_bin" type="pear-config" />
