--- a/HomerMultimedia/src/RTP.cpp
+++ a/HomerMultimedia/src/RTP.cpp
@@ -606,7 +606,6 @@
     mRtpEncoderStream->priv_data = NULL;
     // create monotone timestamps
     mRtpEncoderStream->cur_dts = 0;
-    mRtpEncoderStream->reference_dts = 0;
 
     // set target coordinates for rtp stream
     snprintf(mRtpFormatContext->filename, sizeof(mRtpFormatContext->filename), "rtp://%s:%u", pTargetHost.c_str(), pTargetPort);
