diff --git a/Makefile b/Makefile
index 989ed8c..e387695 100644
--- a/Makefile
+++ b/Makefile
@@ -32,12 +32,12 @@ clean:
 
 install: all
 	$(INSTALL) -m 755 -o root -g root -d \
-		$(DESTDIR)$(PREFIX)/lib \
-		$(DESTDIR)$(PREFIX)/lib/pkgconfig \
+		$(DESTDIR)$(PREFIX)/lib64 \
+		$(DESTDIR)$(PREFIX)/lib64/pkgconfig \
 		$(DESTDIR)$(PREFIX)/include/bmusb
-	$(INSTALL) -m 755 -o root -g root $(LIB) $(SOLIB) $(DESTDIR)$(PREFIX)/lib
-	$(LN) -sf $(SOLIB) $(DESTDIR)$(PREFIX)/lib/$(SONAME)
-	$(LN) -sf $(SOLIB) $(DESTDIR)$(PREFIX)/lib/$(SODEV)
+	$(INSTALL) -m 755 -o root -g root $(LIB) $(SOLIB) $(DESTDIR)$(PREFIX)/lib64
+	$(LN) -sf $(SOLIB) $(DESTDIR)$(PREFIX)/lib64/$(SONAME)
+	$(LN) -sf $(SOLIB) $(DESTDIR)$(PREFIX)/lib64/$(SODEV)
 	$(INSTALL) -m 755 -o root -g root bmusb/bmusb.h bmusb/fake_capture.h $(DESTDIR)$(PREFIX)/include/bmusb
-	$(INSTALL) -m 644 -o root -g root bmusb.pc $(DESTDIR)$(PREFIX)/lib/pkgconfig
+	$(INSTALL) -m 644 -o root -g root bmusb.pc $(DESTDIR)$(PREFIX)/lib64/pkgconfig
 
