diff --git a/scripts/Makefile b/scripts/Makefile
index ac345057..c4a033b2 100644
--- a/scripts/Makefile
+++ b/scripts/Makefile
@@ -100,7 +100,6 @@ test_pl: $(PL_CHECKS)
 	perl -I ../lib -c $<; \
 
 test_py: $(VERSION_FILE)
-	$(foreach python,$(shell py3versions -r ../debian/control),$(python) setup.py test$(\n))
 
 debbisect.1: debbisect
 	help2man \
@@ -151,7 +150,6 @@ clean:
 test: test_pl test_sh test_py
 
 install: all
-	python3 setup.py install --root="$(DESTDIR)" --no-compile --install-layout=deb
 	cp $(SCRIPTS) $(DESTDIR)$(BINDIR)
 	ln -sf edit-patch $(DESTDIR)$(BINDIR)/add-patch
 	install -d $(DESTDIR)$(COMPL_DIR)
