From a2a43dddc1ca73c18724026a9930eca690cb6862 Mon Sep 17 00:00:00 2001
From: Anton Schubert <ischluff@mailbox.org>
Date: Fri, 25 Oct 2019 21:43:23 +0200
Subject: [PATCH] fix newer CEF versions

---
 nageru/cef_capture.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/nageru/cef_capture.cpp b/nageru/cef_capture.cpp
index 9524861..b9baa0e 100644
--- a/nageru/cef_capture.cpp
+++ b/nageru/cef_capture.cpp
@@ -189,7 +189,7 @@ void CEFCapture::start_bm_capture()
 
 		CefWindowInfo window_info;
 		window_info.SetAsWindowless(0);
-		browser = CefBrowserHost::CreateBrowserSync(window_info, cef_client, start_url, browser_settings, nullptr);
+		browser = CefBrowserHost::CreateBrowserSync(window_info, cef_client, start_url, browser_settings, nullptr, nullptr);
 		for (function<void()> &task : deferred_tasks) {
 			task();
 		}
-- 
2.23.0

